crates/oxc_linter/src/rules/eslint/no_loss_of_precision.rs

```diff
@@ -0,0 +1,298 @@
+            format!("{value:x}")
+        } else {
+            format!("{value:o}")
+        };
+        !raw.ends_with(&suffix.to_uppercase())
+    }
+
+    fn base_ten_loses_precision(node: &'_ NumberLiteral) -> bool {
+        let normalized_raw_num = Self::normalize(&Self::get_raw(node));
+        let precision = normalized_raw_num.split('e').next().unwrap().replace('.', "").len();
+
+        if precision > 100 {
+            return true;
+        }
+
+        let stored_num = format!("{:1$}", node.value, precision);
+        let normalized_stored_num = Self::normalize(&stored_num);
+        normalized_raw_num != normalized_stored_num
+    }
+
+    fn add_decimal_point(num: &str) -> String {
+        format!("{}.{}", &num[..1], &num[1..])
+    }
+
+    fn normalize_int(num: &str) -> NormalizedNum {
+        // specially deal with 0
+        if num == "0" {
+            return NormalizedNum { magnitude: 0, coefficient: String::from("0.") };
+        }
+
+        let significant_digits = num.trim_start_matches('0').trim_end_matches('0');
+        #[allow(clippy::cast_possible_wrap)]
+        // the length of number is larger then isize is almost impossible in real-world codebase
+        let magnitude =
+            if num.starts_with('0') { num.len() as isize - 2 } else { num.len() as isize - 1 };
+        NormalizedNum { magnitude, coefficient: Self::add_decimal_point(significant_digits) }
+    }
+
+    fn normalize_float(num: &str) -> NormalizedNum {
+        let trimmed_float = match num.trim_start_matches('0') {
+            "" => "0",
+            s => s,
+        };
+
+        return trimmed_float.strip_prefix('.').map_or_else(
+            || {
+                let trimmed_float = trimmed_float.trim_end_matches('0');
+                // unwrap here will never panic, we guarantee the input contains a `.`
+                #[allow(clippy::cast_possible_wrap)]
+                let magnitude = (trimmed_float.find('.').unwrap() - 1) as isize;
+                NormalizedNum {
+                    coefficient: Self::add_decimal_point(&trimmed_float.replace('.', "")),
+                    magnitude,
+                }
+            },
+            |stripped| {
+                let decimal_digits = stripped;
+                let significant_digits = match decimal_digits.trim_start_matches('0') {
+                    "" => "0",
+                    s => s,
+                };
+
+                #[allow(clippy::cast_possible_wrap)]
+                // the length of number is larger then isize is almost impossible in real-world codebase
+                let magnitude =
+                    significant_digits.len() as isize - decimal_digits.len() as isize - 1;
+                NormalizedNum {
+                    coefficient: Self::add_decimal_point(significant_digits),
+                    magnitude,
+                }
+            },
+        );
+    }
+
+    fn normalize(num: &str) -> String {
+        if num == "inf" {
```

Comment:
inf? I see no test cases for this, did you mean `Infinity`?

Reply:
> inf? I see no test cases for this, did you mean `Infinity`?

I encountered a case that `node.raw` is `inf` when I implementing the lint, however I can't reproduce that case, seems that I misread some dbg print😢. I have also checked eslint, it does not count `Infinity` as loss of precision, I will add it to the test case.